diff --git a/dom/workers/WorkerNavigator.cpp b/dom/workers/WorkerNavigator.cpp
index 622724b529..c9a65ff013 100644
--- a/dom/workers/WorkerNavigator.cpp
+++ b/dom/workers/WorkerNavigator.cpp
@@ -5,6 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/dom/WorkerNavigator.h"
+#include "MaskConfig.hpp"
 
 #include <utility>
 
@@ -103,6 +104,9 @@ JSObject* WorkerNavigator::WrapObject(JSContext* aCx,
 }
 
 bool WorkerNavigator::GlobalPrivacyControl() const {
+  if (auto value = MaskConfig::GetBool("navigator.globalPrivacyControl");
+      value.has_value())
+    return value.value();
   bool gpcStatus = StaticPrefs::privacy_globalprivacycontrol_enabled();
   if (!gpcStatus) {
     JSObject* jso = GetWrapper();
@@ -125,6 +129,8 @@ void WorkerNavigator::SetLanguages(const nsTArray<nsString>& aLanguages) {
 void WorkerNavigator::GetAppVersion(nsString& aAppVersion,
                                     CallerType aCallerType,
                                     ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetString("navigator.appVersion"))
+    return aAppVersion.Assign(NS_ConvertUTF8toUTF16(value.value()));
   WorkerPrivate* workerPrivate = GetCurrentThreadWorkerPrivate();
   MOZ_ASSERT(workerPrivate);
 
@@ -147,6 +153,8 @@ void WorkerNavigator::GetAppVersion(nsString& aAppVersion,
 
 void WorkerNavigator::GetPlatform(nsString& aPlatform, CallerType aCallerType,
                                   ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetString("navigator.platform"))
+    return aPlatform.Assign(NS_ConvertUTF8toUTF16(value.value()));
   WorkerPrivate* workerPrivate = GetCurrentThreadWorkerPrivate();
   MOZ_ASSERT(workerPrivate);
 
@@ -207,6 +215,8 @@ class GetUserAgentRunnable final : public WorkerMainThreadRunnable {
 
 void WorkerNavigator::GetUserAgent(nsString& aUserAgent, CallerType aCallerType,
                                    ErrorResult& aRv) const {
+  if (auto value = MaskConfig::GetString("navigator.userAgent"))
+    return aUserAgent.Assign(NS_ConvertUTF8toUTF16(value.value()));
   WorkerPrivate* workerPrivate = GetCurrentThreadWorkerPrivate();
   MOZ_ASSERT(workerPrivate);
 
@@ -218,6 +228,8 @@ void WorkerNavigator::GetUserAgent(nsString& aUserAgent, CallerType aCallerType,
 }
 
 uint64_t WorkerNavigator::HardwareConcurrency() const {
+  if (auto value = MaskConfig::GetUint64("navigator.hardwareConcurrency"))
+    return value.value();
   RuntimeService* rts = RuntimeService::GetService();
   MOZ_ASSERT(rts);
 
